lib/bdev/nvme/blockdev_nvme.c

```diff
@@ -502,16 +503,59 @@ probe_cb(void *cb_ctx, const struct spdk_nvme_transport_id *trid,
 }
 
 static void
-blockdev_nvme_timeout_cb(void *cb_arg, struct spdk_nvme_ctrlr *ctrlr,
-			 struct spdk_nvme_qpair *qpair, uint16_t cid)
+abort_cb(void *arg, const struct spdk_nvme_cpl *cpl)
+{
+	/*
+	 * abort completion callback function. Currently, there is nothing to do
+	 * struct spdk_nvme_ctrlr *ctrlr = arg;
+	 */
+}
+
+static void
+blockdev_nvme_timeout_cb(struct spdk_nvme_ctrlr *ctrlr,
+			 struct spdk_nvme_qpair *qpair,
+			 void *cb_arg,
```

Comment:
I see what you mean, but I'd rather not remove the extension of the timeout. The extended timeout covers all possible error cases and acts as watchdog timer in a sense. Relying on the abort completion handler can be tricky, e.g. "command not aborted" may or may not be an error depending on the timing.

We can move the reset due to the extended timeout to the NVMe driver which would allow us to use the "timed out" flag and call the callback only once.